proposals/csharp-9.0/records.md

```diff
@@ -9,13 +9,34 @@ The syntax for a record is as follows:
 ```antlr
 record_declaration
     : attributes? class_modifier* 'partial'? 'record' identifier type_parameter_list?
-      parameter_list? record_base? type_parameter_constraints_clause* record_body
+      record_parameters? record_base? type_parameter_constraints_clause* record_body
     ;
 
 record_base
-    : ':' class_type argument_list?
+    : ':' class_type record_base_arguments?
     | ':' interface_type_list
-    | ':' class_type argument_list? ',' interface_type_list
+    | ':' class_type record_base_arguments? ',' interface_type_list
+    ;
+
+record_base_arguments
+    : '(' argument_list? ')'
+    ;
+
+record_parameters
+    : '(' record_parameter_list? ')'
+    ;
+
+record_parameter_list
+    : record_parameter
+    | record_parameter ',' record_parameter_list
+    ;
+
+record_parameter
+    : attributes? type identifier record_property_name? default_argument?
+    ;
+
+record_property_name
+    : ':' identifier
```

Comment:
wait... this is a thing?!  how did i not know about this?

Reply:
Only as part of the `record_parameter` production. So, the `record_property_name` production indicates that the parameter name of a positional record primary constructor lowers to the property name of the record.

Reply:
Thanks @CyrusNajmabadi for calling attention to this. There is no such thing as controlling the parameter name and property name separately. Records just use a normal parameter list after the record's name.